src/clinical/service-worker-thread/tasks.ts

```diff
@@ -190,18 +190,31 @@ const mapEntityDocuments = (
 // Submitted Data Search Results
 function FilterDonorIdDataFromSearch(donors: Donor[], query: ClinicalSearchQuery) {
   const { donorIds, submitterDonorIds } = query;
+  const useQueriedDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
 
-  const useFilteredDonors = !isEmpty(donorIds) || !isEmpty(submitterDonorIds);
-
-  const filteredDonors = useFilteredDonors
-    ? donors.filter(donor => {
+  const filteredDonors = donors
+    .filter(donor => {
+      if (useQueriedDonors) {
         const { donorId, submitterId } = donor;
         const stringId = `${donorId}`;
```

Comment:
does `donorId` come in as a number?

Reply:
Yep https://github.com/icgc-argo/argo-clinical/blob/develop/src/clinical/clinical-entities.ts